src/bin/pgcopydb/ld_transform.c

```diff
@@ -1880,3 +1923,69 @@ stream_write_value(FILE *out, LogicalMessageValue *value)
 
 	return true;
 }
+
+
+/*
+ *  computeTxnMetadataFilename computes the file path for transaction metadata
+ *  based on its transaction id
+ */
+bool
+computeTxnMetadataFilename(uint32_t xid, const char *dir, char *filename)
+{
+	if (dir == NULL)
+	{
+		log_error("BUG: computeTxnMetadataFilename is called with "
+				  "a NULL transaction directory");
+		return false;
+	}
+
+	if (xid == 0)
+	{
+		log_error("BUG: computeTxnMetadataFilename is called with "
+				  "0 transaction xid");
```

Comment:
Again, value at the end of the line, and let's use %d, xid; even though we know it's zero.